src/framework/theme/components/timepicker/timepicker.directive.ts

```diff
@@ -0,0 +1,349 @@
+ * timepicker-active-color:
+ * timepicker-cell-text-font-size:
+ * timepicker-cell-text-font-weight:
+ * timepicker-basic-color:
+ * timepicker-border-color:
+ * timepicker-border-style:
+ * timepicker-border-width:
+ * timepicker-scrollbar-color:
+ * timepicker-scrollbar-background-color:
+ * timepicker-scrollbar-width:
+ * timepicker-cell-line-height:
+ * timepicker-cell-font-size:
+ * timepicker-single-column-width:
+ * timepicker-multiple-column-width:
+ * timepicker-title-height:
+ * timepicker-title-padding:
+ * timepicker-container-width:
+ * timepicker-container-height:
+ * */
+@Directive({
+  selector: 'input[nbTimepicker]',
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => NbTimePickerDirective),
+    multi: true,
+  }],
+})
+export class NbTimePickerDirective<D> implements AfterViewInit, ControlValueAccessor {
+  /**
+   * Provides timepicker component.
+   * */
+  @Input('nbTimepicker')
+  get timepicker(): NbTimePickerComponent<D> {
+    return this._timePickerComponent;
+  }
+
+  set timepicker(timePicker: NbTimePickerComponent<D>) {
+    this._timePickerComponent = timePicker;
+  }
+
+  /**
+   * NbTimePickerComponent instance passed via input.
+   * @docs-private
+   * */
+  protected _timePickerComponent: NbTimePickerComponent<D>;
+
+  /**
+   * Positioning strategy used by overlay.
+   * @docs-private
+   * */
+  protected positionStrategy: NbAdjustableConnectedPositionStrategy;
+  protected overlayRef: NbOverlayRef;
+  protected overlayOffset = 8;
+  protected destroy$: Subject<void> = new Subject<void>();
+  protected _onChange: (value: string) => void = () => {};
+  protected _onTouched = () => {};
+  /**
+   * Trigger strategy used by overlay.
+   * @docs-private
+   * */
+  protected triggerStrategy: NbTriggerStrategy;
+
+  /**
+   * Returns html input element.
+   * */
+  get input(): HTMLInputElement {
+    return this.hostRef.nativeElement;
+  }
+
+  /**
+   * Determines is timepicker overlay opened.
+   * */
+  get isOpen(): boolean {
+    return this.overlayRef && this.overlayRef.hasAttached();
+  }
+
+  /**
+   * Determines is timepicker overlay closed.
+   * */
+  get isClosed(): boolean {
+    return !this.isOpen;
+  }
+
+  constructor(protected positionBuilder: NbPositionBuilderService,
+              protected hostRef: ElementRef,
+              protected triggerStrategyBuilder: NbTriggerStrategyBuilderService,
+              protected overlay: NbOverlayService,
+              protected cd: ChangeDetectorRef,
+              protected calendarTimeModelService: NbCalendarTimeModelService<D>,
+              protected dateService: NbDateService<D>,
+  ) {
+    this.subscribeOnInputChange();
+  }
+
+  /**
+   * Returns host input value.
+   * */
+  get inputValue(): string {
+    return this.input.value;
+  }
+
+  set setInputValue(value: string) {
+    this.input.value = value;
+  }
+
+  ngAfterViewInit() {
+    this.input.placeholder = this.timepicker.timeFormat;
```

Comment:
Is the placeholder a design requirement? Won't it override user defined placeholder?